src/main.rs

```diff
@@ -266,6 +263,59 @@ async fn main(_spawner: Spawner) {
         gps::set_nmea_output().await;
         gps::tx(b"$PMTK220,100*2F\r\n").await;
 
+        let mut wrapper: W25QSequentialStorage<Bank1, CAPACITY> = W25QSequentialStorage::new(flash);
+
+        map::store_item(
+            &mut wrapper,
+            CONFIG_FLASH_RANGE,
+            NoCache::new(),
+            &mut [0; 1024],
+            &U64Item("id".try_into().unwrap(), 4),
+        )
+        .await
+        .unwrap();
+
+        let board_id: Result<Option<U64Item>, _> = map::fetch_item(
+            &mut wrapper,
+            CONFIG_FLASH_RANGE,
+            NoCache::new(),
+            &mut [0; 41],
+            "id".try_into().unwrap(),
+        )
+        .await;
+
+        let board_id = board_id.unwrap().unwrap().1;
+        let role = match board_id {
+            1 => Role::Ground,
+            2 | 3 => Role::Avionics,
+            _ => Role::Cansat,
+        };
+
+        unsafe { mission::ROLE = role };
+
+        let bmp = if mission::role() != Role::Ground {
+            #[cfg(feature = "target-mini")]
+            {
+                Some({
+                    let streams = StreamsTuple::new(dp.DMA1);
+                    let tx_stream = streams.1;
+                    let rx_stream = streams.0;
+                    let i2c = dp
+                        .I2C1
+                        .i2c((gpio.b.pb8, gpio.b.pb7), 400.kHz(), &clocks)
+                        .use_dma(tx_stream, rx_stream);
+                    BMP388Wrapper::new(i2c, &mut delay)
```

Comment:
need to also call `BMP388::set_power_control` and `BMP388::set_oversampling` to get this to work

Reply:
done